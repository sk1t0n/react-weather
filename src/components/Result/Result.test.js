import React from 'react';
import { render } from '@testing-library/react';
import { Result } from './index';

const defaultProps = {
  isLoading: false,
  error: null,
  temperature: null
};

it('renders correctly with default props', () => {
  const { asFragment } = render(
    <Result
      isLoading={defaultProps.isLoading}
      error={defaultProps.error}
      temperature={defaultProps.temperature}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('renders correctly with isLoading as true', () => {
  const { asFragment } = render(
    <Result
      isLoading={true}
      error={defaultProps.error}
      temperature={defaultProps.temperature}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('renders correctly with error', () => {
  const { asFragment } = render(
    <Result
      isLoading={defaultProps.isLoading}
      error={'error'}
      temperature={defaultProps.temperature}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('renders correctly with temperature', () => {
  const { asFragment } = render(
    <Result
      isLoading={defaultProps.isLoading}
      error={defaultProps.error}
      temperature={{ temp_c: 0, temp_f: 32 }}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
