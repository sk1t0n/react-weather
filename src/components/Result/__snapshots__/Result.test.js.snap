// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default props 1`] = `
<DocumentFragment>
  <div
    class="mt-5"
  />
</DocumentFragment>
`;

exports[`renders correctly with error 1`] = `
<DocumentFragment>
  <div
    class="mt-5"
  >
    <div
      class="fade alert alert-danger show"
      role="alert"
    >
      error
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with isLoading as true 1`] = `
<DocumentFragment>
  <div
    class="mt-5"
  >
    <div
      class="lds-dual-ring"
    />
  </div>
</DocumentFragment>
`;

exports[`renders correctly with temperature 1`] = `
<DocumentFragment>
  <div
    class="mt-5"
  >
    <div
      class="fade alert alert-info show"
      role="alert"
    >
      <p>
        Temperature in degrees:
        <span
          class="float-end"
        >
          0 ℃
        </span>
      </p>
      <p>
        Temperature in fahrenheit:
        <span
          class="float-end"
        >
          32 ℉
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
